### PR TITLE
DOCS-11473 added content for retry mechanism

### PR DESCRIPTION
**Pull request by priyaayyar**
Added content for the retry mechanism in the JWT policy.

### CODE REVIEW

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 1]
```diff
@@ -70,6 +70,12 @@ The JWT is also propagated downstream, and you can access it by using the follow
 
 `#[authentication.properties.jwt]`
 
+=== Retry Mechanism
+
+If you selected JSON Web Key Sets (JWKS) as JWT Key Origin, the policy uses a retry mechanism to handle failed JWKS fetches. Every time the cached JWKS expires or is absent and a new request is initiated, a request is sent to the JWKS service. If the request fails, a sequence of asynchronous requests with an exponential backoff delay are sent to the JWKS service until a new JWKS is obtained and cached again.
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: "a sequence of...is sent":
```suggestion
If you selected JSON Web Key Sets (JWKS) as JWT Key Origin, the policy uses a retry mechanism to handle failed JWKS fetches. Every time the cached JWKS expires or is absent and a new request is initiated, a request is sent to the JWKS service. If the request fails, a sequence of asynchronous requests with an exponential backoff delay is sent to the JWKS service until a new JWKS is obtained and cached again. 
```

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 2]
```diff
@@ -70,6 +70,12 @@ The JWT is also propagated downstream, and you can access it by using the follow
 
 `#[authentication.properties.jwt]`
 
+=== Retry Mechanism
+
+If you selected JSON Web Key Sets (JWKS) as JWT Key Origin, the policy uses a retry mechanism to handle failed JWKS fetches. Every time the cached JWKS expires or is absent and a new request is initiated, a request is sent to the JWKS service. If the request fails, a sequence of asynchronous requests with an exponential backoff delay are sent to the JWKS service until a new JWKS is obtained and cached again. 
+
+Until the recaching completes, the expired JWKS is used to process the request, and a `503` error is returned if no request is found. This enables you to use a cached value in upcoming requests, even if the previous requests failed.
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: see style ref. entry for "complete":
```suggestion
Until the recaching is complete, the expired JWKS is used to process the request, and a `503` error is returned if no request is found. This enables you to use a cached value in upcoming requests, even if the previous requests failed.
```
or
```suggestion
Until the recaching finishes, the expired JWKS is used to process the request, and a `503` error is returned if no request is found. This enables you to use a cached value in upcoming requests, even if the previous requests failed.
```

Comment (kbostrom) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Can you clarify to what "This" refers in "This enables you...": is it the fact that the expired JWKS is used, or that the error is returned, or both?

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 3]
```diff
@@ -82,15 +88,15 @@ For Mule applications, the following parameters are displayed based on whether y
 |===
 | Element | Description | Example
 | JWT Origin | Specifies from where in the request the JWT will be extracted:
-* Bearer Authentication Header
+* HTTP Bearer Authentication Header
 * Custom Expression
 
-If you set it to Bearer Authentication Header, the JWT will be expected as Bearer. +
-If you set this field to *Custom Expression*, a DataWeave Expression returning the token must be provided.
+If you set it to HTTP Bearer Authentication Header, the JWT is expected as Bearer. +
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: To what does "it" refer?
And if you are "setting" it using syntax, then it should be formatted as user-supplied syntax.

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 4]
```diff
@@ -82,15 +88,15 @@ For Mule applications, the following parameters are displayed based on whether y
...
+If you set this field to *Custom Expression*, you must provide a DataWeave Expression returning the token.
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: ditto previous comment about format, `Custom Expression`.

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 5]
```diff
@@ -103,7 +109,7 @@ Ignore this field if you selected *none* as JWT Signing Method.
 
 | JWT Key Origin
 | Specifies where to obtain the key for the Signature validation. +
-You can provide the key in the policy selecting the *Text* option or obtain it from a JWKS. +
+You can provide the key in the policy selecting the *Text* option or obtain it from a *JWKS*. +
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]:
```suggestion
You can provide the key in the policy by selecting the *Text* option or by obtaining it from a JWKS. +
```

Comment (kbostrom) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Question: if JWKS = ...Sets, plural, can we actually refer to "a" JWKS? Is it supposed to be Set, singular?

I see both when I search:

JSON Web Key Sets - Auth0https://auth0.com › Docs › Tokens › JSON Web Tokens
JSON Web Key Sets ... The JSON Web Key Set (JWKS) is a set of keys containing the public keys used to verify any JSON Web Token (JWT) issued by the ...
‎JSON Web Key Set Properties · ‎Locate JSON Web Key Sets · ‎Signing Keys

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 6]
```diff
@@ -123,6 +129,11 @@ Ignore this field if you selected *none* as JWT Signing Method.
 Ignore this field if you selected *none* as JWT Signing Method.
 | This field input is the amount of time, in minutes, during which the policy considers the JWKS valid.
 
+| JWKS Service connection timeout (milliseconds)
+| Sets the maximum time, in milliseconds, to wait for a response when authenticating the Access Token validation endpoint. +
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Should Access Token be lowercase here?

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 7]
```diff
@@ -123,6 +129,11 @@ Ignore this field if you selected *none* as JWT Signing Method.
...
+The default value is set to 10 seconds.
```
Comment (kbostrom) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]:
```suggestion
The default value is 10 seconds.
```

**Review by kbostrom — DISMISSED** [review]
CE

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 2 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 3 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 4 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 6 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 7 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 5 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 1 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

**Review by pablogomez93 — APPROVED** [review]

**Review by kbostrom — APPROVED** [review]
RA

**Pull request merged**